pilot/pkg/networking/plugin/mixer/mixer.go

```diff
@@ -185,22 +184,12 @@ func buildTransport(mesh *meshconfig.MeshConfig, uid attribute) *mccpb.Transport
 		port = mixerMTLSPortNumber
 	}
 
-	res := &mccpb.TransportConfig{
-		CheckCluster:  model.BuildSubsetKey(model.TrafficDirectionOutbound, "", model.Hostname(policy), port),
-		ReportCluster: model.BuildSubsetKey(model.TrafficDirectionOutbound, "", model.Hostname(telemetry), port),
+	return &mccpb.TransportConfig{
+		CheckCluster:            model.BuildSubsetKey(model.TrafficDirectionOutbound, "", model.Hostname(policy), port),
+		ReportCluster:           model.BuildSubsetKey(model.TrafficDirectionOutbound, "", model.Hostname(telemetry), port),
+		NetworkFailPolicy:       &mccpb.NetworkFailPolicy{Policy: mccpb.FAIL_CLOSE},
+		AttributesForMixerProxy: &mpb.Attributes{Attributes: attributes{"source.uid": uid}},
```

Comment:
should there be a test for this?

I suppose os.Getenv("MIXER_NEW_ATTRIBUTES") == "" by default so I suppose there should be a new test to make sure this attribute take effect?

Reply:
If you look at the history, Costin added this condition without actually using the environment variables anywhere. There is only one match for the use of this variable, and it's here. I'm trying to restore to how things were pre-change.

Reply:
That I understand. I am just wondering why adding and removing that condition does not involve fixing a test? It seems to mean we are missing a test for it.

Reply:
There is no tests for this feature. I think the best we have is mixer tests but they are not sufficiently detailed to validate that _all_ of telemetry is present.